airflow/models.py

```diff
@@ -910,16 +910,34 @@ def is_queueable(
         elif self.state == State.QUEUED and not include_queued:
             return False
         # is the task runnable and have its dependencies been met?
-        elif (
-                self.state in State.runnable() and
-                self.are_dependencies_met(
-                    ignore_depends_on_past=ignore_depends_on_past,
-                    flag_upstream_failed=flag_upstream_failed)):
+        elif (self.state in State.runnable() and
+                  self.are_dependencies_met(
+                      ignore_depends_on_past=ignore_depends_on_past,
+                      flag_upstream_failed=flag_upstream_failed) and
+                  not self.is_superceded_by_future()):
             return True
         # anything else
         else:
             return False
 
+    @provide_session
+    def is_superceded_by_future(self, session):
+        if not self.task.only_run_latest:
+            return False
+        logging.info("checking for future")
+        TI = TaskInstance
+        ti = session.query(TI).filter(
+            TI.dag_id == self.dag_id,
+            TI.task_id == self.task_id,
+            TI.state == State.SUCCESS,
+            TI.execution_date > self.execution_date,
+        ).first()
+        if ti:
+            if self.state in State.runnable():
+                self.set_state(State.FUTURE_SUCCEEDED, session)
```

Comment:
Agreed that it's not the cleanest, but I'm not sure how easily it can be extracted. It behaves very similarly to evaluate_trigger_rule() which gets called from are_dependencies_met() in the same chain. Is there a work in progress branch anywhere that shows how state assignments in evaluate_trigger_rule() were brought out of the is_ stack that I could reference? The other change I could make would be to add a flag similar to flag_upstream_failed so that state would only change if the original caller explicitly passed through the flag enabling it.